pkg/framework/proxies.go

```diff
@@ -0,0 +1,249 @@
+package framework
+
+import (
+	"context"
+	"errors"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mapiv1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const squidConf = `acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+acl crcnet src 10.88.0.0/16	# RFC1918 possible internal network
+acl SSL_ports port 443
+acl Safe_ports port 80		# http
+acl Safe_ports port 21		# ftp
+acl Safe_ports port 443		# https
+acl Safe_ports port 70		# gopher
+acl Safe_ports port 210		# wais
+acl Safe_ports port 1025-65535	# unregistered ports
+acl Safe_ports port 280		# http-mgmt
+acl Safe_ports port 488		# gss-http
+acl Safe_ports port 591		# filemaker
+acl Safe_ports port 777		# multiling http
+acl CONNECT method CONNECT
+http_access deny !Safe_ports
+http_access deny CONNECT !SSL_ports
+http_access allow localhost manager
+http_access deny manager
+acl mysubnet src 192.168.0.0/16
+http_access allow localhost localnet crcnet
+http_access allow all
+http_port 80
+coredump_dir /var/spool/squid
+refresh_pattern ^ftp:		1440	20%	10080
+refresh_pattern ^gopher:	1440	0%	1440
+refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
+refresh_pattern (Release|Packages(.gz)*)$      0       20%     2880
+refresh_pattern .		0	20%	4320`
+
+// CreateProxyMachineSet creates a machineset that results in a single node being spun up. This node will not be in the worker MCP.
+func CreateProxyMachineSet(client runtimeclient.Client) (*mapiv1beta1.MachineSet, error) {
+	var machineSet *mapiv1beta1.MachineSet
+	var machineSetParams MachineSetParams
+
+	machineSetParams = BuildMachineSetParams(client, 1)
+	machineSetParams.Labels["proxy-host"] = ""
+	machineSetParams.Labels["node-role.kubernetes.io/proxy"] = ""
+	machineSet, err := CreateMachineSet(client, machineSetParams)
+	if err != nil {
+		return nil, err
+	}
+	WaitForMachineSet(client, machineSet.GetName())
+
+	nodes, err := GetNodesFromMachineSet(client, machineSet)
+	if len(nodes) == 0 {
+		return nil, errors.New("No nodes found in machineset")
+	}
+	node := nodes[0]
+	delete(node.Labels, "node-role.kubernetes.io/worker")
+	client.Update(context.Background(), node)
```

Comment:
Should we not check this error?
```suggestion
	if err := client.Update(context.Background(), node); err != nil {
		return nil, err
	}
```